okhttp/src/main/java/okhttp3/internal/http2/Hpack.java

```diff
@@ -463,29 +464,37 @@ void writeHeaders(List<Header> headerBlock) throws IOException {
         smallestHeaderTableSizeSetting = Integer.MAX_VALUE;
         writeInt(maxDynamicTableByteCount, PREFIX_5_BITS, 0x20);
       }
-      // TODO: implement index tracking
-      for (int i = 0, size = headerBlock.size(); i < size; i++) {
-        Header header = headerBlock.get(i);
+
+      for (Header header : headerBlock) {
```

Comment:
Switch back to indexed for here? We try to use them everywhere in libraries to avoid the allocation and indirection that comes with `Iterator`.